webapp/channels/webpack.config.js

```diff
@@ -150,6 +150,7 @@ var config = {
     plugins: [
         new webpack.ProvidePlugin({
             process: 'process/browser.js',
+            Buffer: ['buffer', 'Buffer'],
```

Comment:
pbkdf2 after version 3.1.3 won't auto polyfill buffer (Buffer; provided by Node js buffer module) when used in browser. So we had to make Buffer globally available to have the library working fine in the browser.

Reply:
Just a thought: The only place I see this used is [here](https://github.com/mattermost/mattermost/blob/master/webapp/channels/src/utils/contact_support_sales.ts#L66), I'm wondering if we should replace that with something else.

Reply:
Looking around, browsers surprisingly don't make it easy to convert to and from base 64. There's `btoa` and `atob` which convert between ASCII and base64, but that doesn't work for Unicode characters. There's a script on the MDN which uses `TextEncoder` and some magic to do that properly though which would let us get rid of that polyfill.

https://developer.mozilla.org/en-US/docs/Web/API/Window/btoa#unicode_strings